include/llbuild/Basic/Utility.h

```diff
@@ -0,0 +1,35 @@
+//===- Utility.h ------------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_BASIC_UTILITY_H
+#define LLBUILD_BASIC_UTILITY_H
+
+#include "llvm/ADT/StringRef.h"
+#include <string>
+
+using namespace llvm;
+
+namespace llbuild {
+namespace basic {
+
+/// Escapes control characters in a string to make them displayable on terminal.
+/// For e.g. hello\nworld -> hello\\nworld, input A -> input\ A
```

Comment:
To be a little more explicit, this escapes the string for parsing by the shell.

Note that the `\\n` example isn't actually correct, most shells don't actually honor those kind of escape sequences. Using spaces or characters like `$` as the examples is probably better.